pkg/options/options.go

```diff
@@ -57,8 +67,126 @@ type GCPAPIKey struct {
 }
 
 type AzureAPIKey struct {
-	BaseDnsDomain        string `yaml:"baseDnsDomain"`
-	BaseDomainRGN        string `yaml:"azureBaseDomainRGN"`
-	ServicePrincipalJson string `yaml:"azureServicePrincipalJson"`
-	Region               string `yaml:"region"`
+	BaseDnsDomain  string `yaml:"baseDnsDomain"`
+	BaseDomainRGN  string `yaml:"azureBaseDomainRGN"`
+	ClientId       string `yaml:"clientId"`
+	ClientSecret   string `yaml:"clientSecret"`
+	TenantId       string `yaml:"tenantId"`
+	SubscriptionId string `yaml:"subscriptionId"`
+	Region         string `yaml:"region"`
+}
+
+const charset = "abcdefghijklmnopqrstuvwxyz" +
+	"0123456789"
+
+var TestOptions TestOptionsT
```

Comment:
TestOptions is a global var? and it can only have one? any special reason of designing like this? I can understand that normally user will only have one config for e2e, but feel weird as this is a general library for user to load options? at least we should let user know they can only load one options at a time.

Also for functions, they are using the global var `TestOptions` directly. If we will allow multiple `TestOptions` to be loaded, we can modify functions for `TestOptions` type like func (TestOptions* t) GetHubKubeconfig()?

Reply:
Yes it was designed as only one option file per test... No requirements were made to have multiple options file per running ginkgo running instance.